### ISSUE TITLE
BUG: Embed not rendering template correctly

### ISSUE THREAD

**Issue by carbontwelve** [photogabble/eleventy-plugin-interlinker#66]
11ty will typically pass a markdown file through liquid either before or after markdown-it has had a go (I forget which order) while this plugin seems to only pass it through markdown-it. This means that any shortcodes will not get run resulting in incorrect output.

**Comment by carbontwelve** [comment]
I have written a test to check this and it passes, meaning that short codes are being ran before the embed is embedded.

**Comment by carbontwelve** [comment]
![Image](https://github.com/user-attachments/assets/cb27698d-ee5c-4417-b056-ef17085f08c4)

After some tinkering, I realised that I was only testing using the default template that `defaultEmbedFn` uses when no embed templates is set:

```js
  const tpl = layout === null
    ? template.content
    : `{% layout "${layout}" %} {% block content %} ${template.content} {% endblock %}`;
```

When it's using `template.content` it works as expected, however when `tpl` is set to use the defined layout it fails to run correctly:

```liquid
{% sc %}
Hello world
{% endsc %}
```

VS

```liquid
{% layout "_layouts/test-embed.liquid" %} {% block content %} 
{% sc %}
Hello world
{% endsc %}
 {% endblock %}
```

I wonder if there is something stopping it running the shortcode when it's within a block section - there shouldn't be, should there?

**Comment by carbontwelve** [comment]
I can confirm that this was fixed by #67 and can see it working as expected in my PhotoGabble Website.
